pkg/controller/servicebinding/reconciler_test.go

```diff
@@ -641,10 +666,39 @@ func TestBindTwoSbrsWithSingleApplication(t *testing.T) {
 		err = runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, dep)
 		require.NoError(t, err)
 
+		var envList []corev1.EnvFromSource
+
 		// Assert SBR name with secretRef object in containers env var
 		// Expected secretRef=binding-request1
-		require.Equal(t, sbrName1, dep.Spec.Template.Spec.Containers[0].EnvFrom[0].SecretRef.LocalObjectReference.Name)
+		bindingSecret1, err := fakeDynClient.Resource(secretsGVR).Get(sbrOutput1.Status.Secret, metav1.GetOptions{})
+		require.NoError(t, err)
+		require.Equal(t, sbrOutput1.Status.Secret, bindingSecret1.GetName())
+		envList = append(envList, corev1.EnvFromSource{
+			ConfigMapRef: nil,
+			SecretRef: &corev1.SecretEnvSource{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: bindingSecret1.GetName(),
+				},
+			},
+		})
+
 		// Expected secretRef=binding-request2
-		require.Equal(t, sbrName2, dep.Spec.Template.Spec.Containers[0].EnvFrom[1].SecretRef.LocalObjectReference.Name)
+		bindingSecret2, err := fakeDynClient.Resource(secretsGVR).Get(sbrOutput2.Status.Secret, metav1.GetOptions{})
+		require.NoError(t, err)
+		require.Equal(t, sbrOutput2.Status.Secret, bindingSecret2.GetName())
+
+		envList = append(envList, corev1.EnvFromSource{
+			ConfigMapRef: nil,
+			SecretRef: &corev1.SecretEnvSource{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: bindingSecret2.GetName(),
+				},
+			},
+		})
+
+		require.Contains(t, dep.Spec.Template.Spec.Containers[0].EnvFrom, envList[0])
```

Comment:
This should be an equality check instead, and `envList` should contain the expected items in the expected order:

```go
require.Equal(t, dep.Spec.Template.Spec.Containers[0].EnvFrom, envList)
```